wasm/wat/convert.go

```diff
@@ -25,7 +25,11 @@ func TextToBinary(source []byte) (result *wasm.Module, err error) {
 	// difference is that the text format has type names and the binary format does not.
 	result = &wasm.Module{}
 	for _, t := range m.types {
-		result.TypeSection = append(result.TypeSection, &wasm.FunctionType{Params: t.params, Results: t.results})
+		var results []wasm.ValueType
+		if t.result != 0 {
```

Comment:
Since we are enforcing only one result per signature in the parser anyway, I made the tracking simpler by using a single value instead.